engine/src/main/java/org/camunda/bpm/engine/impl/ManagementServiceImpl.java

```diff
@@ -513,6 +555,49 @@ public TelemetryData getTelemetryData() {
     return commandExecutor.execute(new GetTelemetryDataCmd());
   }
 
+  /**
+   * Adds the web application name to the telemetry data of the engine.
+   *
+   * @param webapp
+   *          the web application that is used with the engine
+   * @return whether the web application was successfully added or not
+   */
+  public boolean addWebappToTelemetry(String webapp) {
+    TelemetryRegistry telemetryRegistry = processEngineConfiguration.getTelemetryRegistry();
+    if (telemetryRegistry != null) {
+      telemetryRegistry.addWebapp(webapp);
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Adds the application server information to the telemetry data of the engine.
+   *
+   * @param appServerInfo
+   *          a String containing information about the application server
+   */
+  public void addApplicationServerInfoToTelemetry(String appServerInfo) {
+    TelemetryRegistry telemetryRegistry = processEngineConfiguration.getTelemetryRegistry();
+    if (telemetryRegistry != null) {
+      telemetryRegistry.setApplicationServer(appServerInfo);
+    }
+  }
+
+  /**
+   * Adds license key information to the telemetry data of the engine.
+   *
+   * @param licenseKeyData
+   *          a data object containing various pieces of information
+   *          about the installed license
+   */
+  public void addLicenseKeyToTelemetry(LicenseKeyDataImpl licenseKeyData) {
```

Comment:
Since we are also using this method to delete the license key, I think `set` instead of `add` would be a more appropriate verb.